Reply/app/src/main/res/layout/fragment_email.xml

```diff
@@ -16,128 +16,205 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto">
 
-    <data>
-        <variable
-            name="email"
-            type="com.materialstudies.reply.data.Email" />
-    </data>
-
-    <androidx.core.widget.NestedScrollView
-        android:id="@+id/nested_scroll_view"
+  <data>
+    <variable
+        name="email"
+        type="com.materialstudies.reply.data.Email" />
+  </data>
+
+  <androidx.constraintlayout.widget.ConstraintLayout
+      android:layout_width="match_parent"
+      android:layout_height="match_parent"
+      android:layout_marginStart="@dimen/grid_2"
+      android:layout_marginEnd="@dimen/grid_2"
+      android:layout_marginHorizontal="@dimen/grid_0_5"
+      android:layout_marginVertical="@dimen/grid_4">
+
+    <com.google.android.material.appbar.AppBarLayout
+        android:id="@+id/email_app_bar_layout"
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:fillViewport="true"
-        android:paddingHorizontal="@dimen/grid_0_5"
-        android:paddingTop="@dimen/grid_1"
-        android:clipToPadding="false"
-        android:background="?android:colorBackground"
-        app:paddingTopSystemWindowInsets="@{true}">
-
-        <com.google.android.material.card.MaterialCardView
-            android:id="@+id/email_card_view"
+        android:layout_height="wrap_content"
+        android:paddingTop="@dimen/grid_2"
+        android:paddingBottom="@dimen/grid_2"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+      <androidx.appcompat.widget.Toolbar
```

Comment:
You can get rid of this entire <Toolbar> view and all of its children and replace it with the following:


```
      <com.google.android.material.appbar.MaterialToolbar
          android:id="@+id/toolbar"
          android:layout_width="match_parent"
          android:layout_height="wrap_content"
          app:title="@{email.subject}"
          app:titleTextAppearance="?attr/textAppearanceBodyExtraLarge"
          app:titleCentered="true"
          app:subtitle="@{email.messageCountDesc}"
          app:subtitleTextAppearance="?attr/textAppearanceLabelMedium"
          app:subtitleCentered="true"
          app:navigationIcon="@drawable/ic_arrow_back_protected"
          app:menu="@menu/email_overflow_menu" />
```
The `titleTextAppearance` is `textAppearanceBodyExtraLarge`. That doesn't exist in the typography system, but I think you should create a custom attr for it and update `TextAppearance.Reply.EmailSubject` to be `TextAppearance.Reply.BodyExtraLarge` and set the custom theme attr to that style. It makes it feel like you're extending the existing type system instead of creating a one-off style.



`@menu/email_overflow_menu`
Just duplicate the the existing `@menu/email_bottom_sheet_menu` and change all of the item's `showAsAction` property to `never`. This will make it so that the toolbar will only show a 3 dot overflow icon to access the items. That also means you can delete the `ic_triple_vert_dots` drawable you added.
```
...
<menu xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto">

        <item
            android:id="@+id/menu_forward"
            android:icon="@drawable/ic_forward"
            app:showAsAction="never"
            android:title="@string/menu_item_forward"/>

        <item
            android:id="@+id/menu_reply"
            android:icon="@drawable/ic_reply"
            app:showAsAction="never"
            android:title="@string/menu_item_reply"/>

        <item
            android:id="@+id/menu_reply_all"
            android:icon="@drawable/ic_reply_all"
            app:showAsAction="never"
            android:title="@string/menu_item_reply_all"/>

        <item
            android:id="@+id/menu_archive"
            android:icon="@drawable/ic_archive"
            app:showAsAction="never"
            android:title="@string/menu_item_archive"/>

        <item
            android:id="@+id/menu_delete"
            android:icon="@drawable/ic_delete"
            app:showAsAction="never"
            android:title="@string/menu_item_delete"/>
</menu>
```



`@drawable/ic_arrow_back_protected`
You can create this drawable that includes a circular background and a centered icon for the Toolbar's navigation icon. The oval's color should be ?attr/colorSurface, but you wouldn't be able to see that until we update the window background to be an elevated surface color. So maybe add a TODO to this file to remember to update the icon color. Alternatively, just update it now but know it will be invisible since it will match the current background. I would also pull out the width and height values into dimens instead of hardcoding them like in the below snippet.
```
<?xml version="1.0" encoding="utf-8"?>
<ripple
    xmlns:android="http://schemas.android.com/apk/res/android"
    android:color="?attr/colorControlNormal">
    <item>
        <layer-list>
            <!-- Shape background -->
            <item>
                <shape android:shape="oval">
                    <size android:height="44dp" android:width="44dp"/>
                    <solid android:color="?attr/colorPrimaryContainer"/>
                </shape>
            </item>
            <!-- Icon foreground -->
            <item
                android:drawable="@drawable/ic_arrow_back"
                android:width="20dp"
                android:height="20dp"
                android:gravity="center"/>
        </layer-list>
    </item>
</ripple>
```